### PR TITLE
Add polochonfs

### PR DESCRIPTION
**Pull request by gregdel**

### CODE REVIEW

File: lib/papi/client.go [hunk 1]
```diff
@@ -138,6 +139,7 @@ func (c *Client) request(httpType, url string, data io.Reader, result interface{
 		return err
 	}
 	req.Header.Add("User-Agent", "papi")
+	req.Header.Add("Accept-Encoding", "gzip")
```
Comment (PouuleT) [inline lib/papi/client.go:142]: Did you try *not* to manually add the accept-encoding header?
I think that de default http golang behaviour is to accept gzip. And if the request (which gzip was "automatically" added) gets a gzip, response, the transport will transparently handle the decompression
https://go.dev/src/net/http/transport.go#L194

File: app/server/movies.go [hunk 2]
```diff
@@ -64,6 +57,38 @@ func (s *Server) serveMovie(w http.ResponseWriter, req *http.Request) {
...
+	var path string
+	for _, f := range []struct {
+		suffix, path string
+	}{
+		{suffix: ".nfo", path: movie.NfoPath()},
+		{suffix: "poster.jpg", path: movie.MovieThumbPath()},
+		{suffix: "fanart.jpg", path: movie.MovieFanartPath()},
+	} {
+		if strings.HasSuffix(name, f.suffix) {
+			path = f.path
```
Comment (PouuleT) [inline app/server/movies.go:80]: We could break here?
```suggestion
			path = f.path
			break
```

File: lib/papi/show.go [hunk 3]
```diff
@@ -76,10 +91,13 @@ func extractSeasons(imdbID string, input map[string]map[string]*index.Episode) (
 				subs = nil
 			}
 
-			s.Episodes[en] = &Episode{
+			ne := &Episode{
 				ShowEpisode: pe,
 				Subtitles:   subs,
 			}
+
+			s.Episodes[en] = ne
```
Comment (PouuleT) [inline lib/papi/show.go:99]: Most confusing naming ever :p

File: cmd/polochonfs/fs.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+		"\033[33mFUSE\033[0m ", // Yellow prefix
+		logger.Ldate|logger.Ltime,
+	)
+	return fs.Mount(pfs.mountPoint, pfs.root, &fs.Options{
+		RootStableAttr: &fs.StableAttr{Ino: baseInode},
+		UID:            uid,
+		GID:            gid,
+		Logger:         customLogger,
+		MountOptions: fuse.MountOptions{
+			// Enforce sequencial read, one read at a time. This is usefull to
```
Comment (PouuleT) [inline cmd/polochonfs/fs.go:115]:
```suggestion
			// Enforce sequencial read, one read at a time. This is useful to
```

File: lib/papi/client.go (hunk 1 above)
Comment (gregdel) [reply lib/papi/client.go:142]: You're absolutely right, there's nothing to do on the client side: no header, no decoding of the body. Everything is done automatically.
Fixed !

File: app/server/movies.go (hunk 2 above)
Comment (gregdel) [reply app/server/movies.go:80]: You're right, we can even simplify and make this code more correct by matching the exact file name:
```go
var path string
for _, p := range []string{
    movie.NfoPath(),
    movie.MovieThumbPath(),
    movie.MovieFanartPath(),
} {
    if name == filepath.Base(p) {
        path = p
        break
    }
}
```

File: lib/papi/file.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
+package papi
+
+import (
+	index "github.com/odwrtw/polochon/lib/media_index"
+)
+
```
[the remark this answers is not in the document]
Comment (gregdel) [reply lib/papi/file.go:6]: You're right, my setup is fucked....

File: go.mod [hunk 6]
```diff
@@ -34,6 +34,8 @@ require (
 	gopkg.in/yaml.v2 v2.4.0
 )
 
+require github.com/hanwen/go-fuse/v2 v2.5.1
```
[the remark this answers is not in the document]
Comment (gregdel) [reply go.mod]: Nope, fixed

File: cmd/polochonfs/node.go [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+		l.Debug("Context cancelled")
+	default:
+		l.WithField("error", err).Error("Failed to read from async reader")
+	}
+
+	fh.close()
+	return fuse.ReadResultData(dest), defaultErr
+}
+
+func fuseError(err error) syscall.Errno {
```
Comment (PouuleT) [inline cmd/polochonfs/node.go:375]: Not used?

Comment (gregdel) [reply cmd/polochonfs/node.go]: Removed

**Pull request merged**